pennylane/capture/transforms.md

```diff
@@ -0,0 +1,370 @@
+```
+
+```python
+def bind(prim, *args, **params):
+    print("Call bind")
+    top_trace = find_top_trace(args)
+    tracers = [full_raise(top_trace, arg) for arg in args]
+    outs = top_trace.process_primitive(prim, tracers, params)
+    return [full_lower(out) for out in outs]
+```
+
+Finally, let's define a minimal version of the JAX `Primitive` to integrate with our framework:
+
+```python
+from typing import Callable
+
+class Primitive:
+    name: str
+    impl: Callable
+    abstract_eval: Callable
+
+    def __init__(self, name: str):
+        self.name = name
+
+    def def_impl(self, impl):
+        self.impl = impl
+        return self.impl
+
+    def def_abstract_eval(self, abstract_eval):
+        self.abstract_eval = abstract_eval
+        return self.abstract_eval
+
+    def bind(*args, **kwargs):
+        return bind(*args, **kwargs)
+```
+
+Now, we have everything we need to start implementing interpreters. Let's start with the level-0
+interpreter, called the Evaluation Interpreter.
+
+###  Evaluation Interpreter
+
+```python
+class EvalTrace(Trace):
+    pure = lift = lambda self, x: x  # no boxing in Tracers needed
+
+    def process_primitive(self, primitive, tracers, params):
+        """Just do standard evaluation of the primitive being binded (bound?)"""
+        print("Binding primitive with standard evaluation")
+        return [primitive.impl(*tracers, **params)]
+```
+
+With this interpreter, we can evaluate functions. Let's try it out. First, we define a simple
+`Primitive`:
+
+```python
+def sin(x):
+    prim = Primitive("sin")
+
+    @prim.def_impl
+    def _(*args, **kwargs):
+        return np.sin(args[0])
+
+    @prim.def_abstract_eval
+    def _(*args, **kwargs):
+        return jax.core.ShapedArray(args[0].shape, args[0].dtype)
+
+    return prim.bind(x)[0]
+```
+
+Now, let's create a simple user function that uses our `sin` `Primitive` and bring it all together:
+
+```python
+def f(x):
+    return sin(x)
+```
+```pycon
+>>> print(f(3.5))
+Call bind
+Binding primitive with standard evaluation
+-0.35078322768961984
+>>> print(np.sin(3.5))
+-0.35078322768961984
+```
+
+Now that we have a good understanding of how JAX's core works, we can start creating `Trace`s for
+our purposes.
+
+## PennyLane Transforms
+
+As mentioned earlier, JAX has a built in optimization to avoid processing primitives with
+unnecessary interpreters by leveraging data-dependence: we only apply an interpreter when
+the input arguments to a `primitive.bind` call are boxed in a `Tracer` corresponding to
+the interpreter. However, this optimization is unfavourable for our purposes. We want to
+interpret _all_ operator and measurement primitives using our "pennylane transform
+interpreter", and we cannot enforce the boxing of inputs to these primitives inside custom
+`Tracer`s.
+
+To circumvent this, we can use a `PlxprInterpreter` to selectively box inputs of operator and
```

Comment:
```suggestion
To circumvent this, we can use a `PlxprInterpreter` to selectively box inputs of operators and
```

Reply:
The full phrase is "operator and measurement primitives", so I don't think this needs to change 🙂